numpy/core/arrayprint.py

```diff
@@ -521,7 +515,9 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
-    if a.size == 0:
+    if style is not None and a.shape == () and a.dtype.names is None:
+        return style(a[()])
```

Comment:
Worth adding a comment here that the `names is None` check is to avoid co-recursion with `voidtype_str`, which upcasts scalars to arrays again